presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/credential/file/FileCredentialFactory.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc.credential.file;
+
+import io.airlift.configuration.ConfigurationFactory;
+import io.prestosql.plugin.jdbc.credential.CredentialConfig;
+import io.prestosql.plugin.jdbc.credential.CredentialProvider;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static io.airlift.configuration.ConfigurationLoader.loadPropertiesFrom;
+import static java.util.Objects.requireNonNull;
+
+public class FileCredentialFactory
+        implements Provider<CredentialProvider>
+{
+    private final CredentialConfig credentialsConfig;
+
+    @Inject
+    public FileCredentialFactory(FileCredentialProviderConfig config)
+            throws IOException
+    {
+        requireNonNull(config, "config is null");
+        Map<String, String> properties = loadPropertiesFrom(config.getCredentialFile());
+        credentialsConfig = new ConfigurationFactory(properties).build(CredentialConfig.class);
+    }
+
+    @Override
+    public CredentialProvider get()
+    {
+        return new FileCredentialProvider(credentialsConfig);
```

Comment:
you could returm `InlineCredentialProvider`.

So maybe ` Provider<CredentialProvider>` you have ` Provider< CredentialConfig>` and then bind `InlineCredentialProvider` to `CredentialProvider`.